server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1708,8 +1739,26 @@ synchronized void addShardDeleteResult(
         ) {
             try {
                 shardGenerationsBuilder.put(indexId, shardId, newGeneration);
-                new ShardSnapshotMetaDeleteResult(Objects.requireNonNull(indexId.getId()), shardId, blobsToDelete).writeTo(compressed);
-                resultCount += 1;
+                // There is a minimum of 1 byte available for writing
+                if (this.truncatedShardDeleteResultsOutputStream.hasCapacity()) {
+                    new ShardSnapshotMetaDeleteResult(Objects.requireNonNull(indexId.getId()), shardId, blobsToDelete).writeTo(compressed);
+                    // We only want to read this shard delete result if we were able to write the entire object.
+                    // Otherwise, for partial writes, an EOFException will be thrown upon reading
+                    if (this.truncatedShardDeleteResultsOutputStream.hasCapacity()) {
+                        successfullyWrittenBlobsCount += 1;
```

Comment:
This replaces `resultCount` but it's the count of the number of successfully recorded _shards_ not _blobs_.